File: src/main/java/com/nanaios/polygonal_tech/capability/impl/LongFluidHandler.java
package com.nanaios.polygonal_tech.capability.impl;

import com.nanaios.polygonal_tech.capability.interfaces.ILongFluidHandler;
import com.nanaios.polygonal_tech.fluids.base.LongFluidStack;
import net.minecraftforge.fluids.FluidStack;
import org.jetbrains.annotations.NotNull;

public class LongFluidHandler implements ILongFluidHandler {
    @Override
    public long getLongFluidAmount() {
        return 0;
    }

    @Override
    public long getLongTankCapacity() {
        return 0;
    }

    @Override
    public long fillLongFluid(LongFluidStack resource, boolean simulate) {
        return 0;
    }

    @Override
    public LongFluidStack drainLongFluid(LongFluidStack resource, boolean simulate) {
        return null;
    }

    @Override
    public LongFluidStack drainLongFluid(long maxDrain, boolean simulate) {
        return null;
    }

    @Override
    public int getTanks() {
        return 0;
    }

    @Override
    public @NotNull FluidStack getFluidInTank(int i) {
        return null;
    }

    @Override
    public int getTankCapacity(int i) {
        return 0;
    }

    @Override
    public boolean isFluidValid(int i, @NotNull FluidStack fluidStack) {
        return false;
    }

    @Override
    public int fill(FluidStack fluidStack, FluidAction fluidAction) {
        return 0;
    }

    @Override
    public @NotNull FluidStack drain(FluidStack fluidStack, FluidAction fluidAction) {
        return null;
    }

    @Override
    public @NotNull FluidStack drain(int i, FluidAction fluidAction) {
        return null;
    }
}

File: src/main/kotlin/com/nanaios/polygonal_tech/lib/capability/fluid/LongFluidHandler.kt
package com.nanaios.polygonal_tech.lib.capability.fluid

import com.nanaios.polygonal_tech.lib.fluid.LongFluidStack
import net.minecraftforge.fluids.capability.IFluidHandler

open class LongFluidHandler(
    capacity: Long
):ILongFluidHandler {
    override fun getLongFluidInTank(tank: Int): LongFluidStack {
        TODO("Not yet implemented")
    }

    override fun getTankLongCapacity(tank: Int): Long {
        TODO("Not yet implemented")
    }

    override fun isFluidValid(
        tank: Int,
        stack: LongFluidStack
    ): Boolean {
        TODO("Not yet implemented")
    }

    override fun fill(
        resource: LongFluidStack,
        action: IFluidHandler.FluidAction
    ): Long {
        TODO("Not yet implemented")
    }

    override fun drain(
        resource: LongFluidStack,
        action: IFluidHandler.FluidAction
    ): LongFluidStack {
        TODO("Not yet implemented")
    }

    override fun drain(
        maxDrain: Long,
        action: IFluidHandler.FluidAction
    ): LongFluidStack {
        TODO("Not yet implemented")
    }

    override fun getTanks(): Int {
        TODO("Not yet implemented")
    }
}

class InputOnlyLongFluidHandler(
    handler:ILongFluidHandler,
): ILongFluidHandler by handler {
    override fun drain(
        resource: LongFluidStack,
        action: IFluidHandler.FluidAction
    ): LongFluidStack {
        return LongFluidStack.EMPTY
    }

    override fun drain(
        maxDrain: Long,
        action: IFluidHandler.FluidAction
    ): LongFluidStack {
        return LongFluidStack.EMPTY
    }
}

class OutputOnlyLongFluidHandler(
    handler:ILongFluidHandler,
): ILongFluidHandler by handler {
    override fun fill(
        resource: LongFluidStack,
        action: IFluidHandler.FluidAction
    ): Long {
        return 0L
    }
}